evalml/pipelines/component_graph.py

```diff
@@ -45,11 +45,15 @@ def __init__(self, component_dict=None, random_seed=0):
                 "component_dict must be a dictionary which specifies the components and edges between components"
             )
         self._validate_component_dict()
+
         self.component_instances = {}
         self._is_instantiated = False
         for component_name, component_info in self.component_dict.items():
             component_class = handle_component_class(component_info[0])
             self.component_instances[component_name] = component_class
+
+        self._validate_component_dict_edges()
```

Comment:
Since this requires instantiation of components (to more easily handle detecting if the component is a subclass of a target transformer that would return a tuple), I had to create a separate helper method that was separate from `self._validate_component_dict()`